options.go

```diff
@@ -103,3 +113,16 @@ func (o initiallyPassing) applyCheck(c *checkConfig) {
 func InitiallyPassing(b bool) Option {
 	return initiallyPassing(b)
 }
+
+type executionTimeout time.Duration
+
+func (o executionTimeout) applyCheck(c *checkConfig) {
+	c.executionTimeout = time.Duration(o)
+}
+
+// ExecutionTimeout sets the timeout of the check.
+// It is up to the check to respect the timeout, which is provided via the Context argument of `Check.Execute` method.
+// Defaults to no timeout
```

Comment:
See the above warning: this is only true if there is no preexisting deadline in the context.